crates/router_env/src/logger/formatter.rs

```diff
@@ -187,7 +192,18 @@ where
             }
         });
 
-        Self {
+        // Check for invalid configuration values in default_fields
+        for key in default_fields.keys() {
+            if !IMPLICIT_KEYS.contains(key.as_str()) {
+                return Err(ConfigError::Message(format!(
+                    "Invalid configuration value for key: {}",
+                    key
+                ))
+                .into());
+            }
+        }
```

Comment:
Since you're calling this after the `retain()` call, you won't have any of the "reserved" keys in `default_fields`, right? 
Can you include a screenshot which shows this error being raised and logged?

Also, on the error message, can we have something like: `A reserved key was included in default keys: {key}`?